src/components/FormCalculator.tsx

```diff
@@ -1,38 +1,54 @@
 import { useState } from "react";
 import { FormInput } from "./FormInput";
-import { calculateAge } from "../helpers/calculateAge";
+import { calculateAge, validateFutureDate } from "../helpers/index";
 import { TDate } from "../types/TDate";
 
 export const FormCalculator = () => {
   
-  const [date, setDate] = useState<TDate | null>(null);
-  const [errors, setErrors] = useState({day:'', month:'', year:''})
-  
-  const handleCalculateAge = () => {  
-    calculateAge(date!);
-  };
+  const [date, setDate] = useState<TDate>({ day: '', month: '', year: ''});
+  const [errors, setErrors] = useState({day:'', month:'', year:''});
 
   const handleDateValue = (property: keyof TDate, value:string) => {
 
-    if(property === 'day') {
-      // Matches numbers 01-31, leading 0 is optional
-      const isValidDay = /^(0?[1-9]|[12][0-9]|3[01])$/.test(value);
-      setErrors( prevError => ( {...prevError, day: !isValidDay ? 'Must be a valid day' : ''} ) );
-    } 
-    if(property === 'month') {
-      // Matches numbers 01-12, leading 0 is optional
-      const isValidMonth = /^(0?[1-9]|1[0-2])$/.test(value);
-      setErrors( prevError => ( {...prevError, month: !isValidMonth ? 'Must be a valid month' : ''} ) );
-    } 
-    if(property === 'year') {
-      // Matches 1 to 4 digits for year
-      const isValidYear = /^\d{1,4}$/.test(value);
-      setErrors( prevError => ( {...prevError, year: !isValidYear ? 'Must be a valid year' : ''} ) );
-    } 
-
-    setDate( oldDate => ( { ...oldDate, [property]: value } as TDate) ); 
+    const validateDate = (dateValue:string, regEx:RegExp, msgError:string) => {
+      const isValidDate = regEx.test(dateValue);
+      
+      if(dateValue === '') {
+        setErrors( prevError => ( {...prevError, [property]: 'This field is required'} ) );
+        return false;
+      }
+      if(!isValidDate) {
+        setErrors( prevError => ( {...prevError, [property]: msgError} ) );
+        return false;
+      }
+      setErrors( prevError => ( {...prevError, [property]: ''} ) );
+      return true;
+    }
+
+    if(property === 'day') validateDate(value, /^(0?[1-9]|[12][0-9]|3[01])$/, 'Must be a valid day'); // RegEx matches numbers 01-31, leading 0 is optional
+
+    if(property === 'month') validateDate(value, /^(0?[1-9]|1[0-2])$/, 'Must be a valid month'); // RegEx matches numbers 01-12, leading 0 is optional
+       
+    if(property === 'year') validateDate(value, /^\d{1,4}$/, 'Must be a valid year'); //RegEx matches 1 to 4 digits for year
```

Comment:
Si le pones el año en plan "98" en vez de "1998" me deja pasar. Desconozco si sea la funcionalidad esperada pero el resultado es el mismo para la clase `Date`. Sin embargo, si pones algo como "123" en el año, puede que te de resultados no esperados. Por ende, recomiendo que el regex siempre valide que hayan 4 caracteres para el año.

```suggestion
    if(property === 'year') validateDate(value, /^\d{4}$/, 'The year must have four characters');
```

Reply:
No había notado ese detalle. Quedó ya solucionado y todas las validaciones se han movido al momento de dar clic y mostrar los errores de esa forma.